packages/moonstone/Scroller/Scroller.js

```diff
@@ -240,36 +280,24 @@ class ScrollerBase extends Component {
 	 * @private
 	 */
 	calculatePositionOnFocus = ({item, scrollPosition}) => {
-		if (!this.uiRefCurrent.isVertical() && !this.uiRefCurrent.isHorizontal() || !item || !this.uiRefCurrent.containerRef.current.contains(item)) {
+		const containerNode = this.uiRefCurrent.containerRef.current;
+		const horizontal = this.uiRefCurrent.isHorizontal();
+		const vertical = this.uiRefCurrent.isVertical();
+
+		if (!vertical && !horizontal || !item || !containerNode.contains(item)) {
 			return;
 		}
 
-		if (this.uiRefCurrent.isVertical()) {
-			this.uiRefCurrent.scrollPos.top = this.calculateScrollTop(item);
-		} else if (this.uiRefCurrent.isHorizontal()) {
-			const {
-				left: itemLeft,
-				width: itemWidth
-			} = this.getFocusedItemBounds(item);
+		const containerRect = getRect(containerNode);
+		const itemRect = getRect(item);
 
-			const
-				{rtl} = this.props,
-				{clientWidth} = this.uiRefCurrent.scrollBounds,
-				rtlDirection = rtl ? -1 : 1,
-				{left: containerLeft} = this.uiRefCurrent.containerRef.current.getBoundingClientRect(),
-				scrollLastPosition = scrollPosition ? scrollPosition : this.uiRefCurrent.scrollPos.left,
-				currentScrollLeft = rtl ? (this.uiRefCurrent.scrollBounds.maxLeft - scrollLastPosition) : scrollLastPosition,
-				// calculation based on client position
-				newItemLeft = this.uiRefCurrent.containerRef.current.scrollLeft + (itemLeft - containerLeft);
-
-			if (newItemLeft + itemWidth > (clientWidth + currentScrollLeft) && itemWidth < clientWidth) {
-				// If focus is moved to an element outside of view area (to the right), scroller will move
-				// to the right just enough to show the current `focusedItem`. This does not apply to
-				// `focusedItem` that has a width that is bigger than `this.scrollBounds.clientWidth`.
-				this.uiRefCurrent.scrollPos.left += rtlDirection * ((newItemLeft + itemWidth) - (clientWidth + currentScrollLeft));
-			} else if (newItemLeft < currentScrollLeft) {
-				// If focus is outside of the view area to the left, move scroller to the left accordingly.
-				this.uiRefCurrent.scrollPos.left += rtlDirection * (newItemLeft - currentScrollLeft);
+		if (!contains(containerRect, itemRect)) {
```

Comment:
Is this condition is necessary? It looks performing some redundant comparison with following two `if` statements.

Reply:
Good catch. That was a redundant guard based on my initial implementation.